src/number_theory/bernoulli.lean

```diff
@@ -139,8 +163,44 @@ begin
   rw ← cast_sub hk,
   congr',
   field_simp [show ((n - k : ℕ) : ℚ) + 1 ≠ 0, by {norm_cast, simp}],
-  -- down to nat
   norm_cast,
   rw [mul_comm, nat.sub_add_eq_add_sub hk],
   exact choose_mul_succ_eq n k,
 end
+
+open power_series
+open nat
+
+@[simp] lemma constant_coeff_exp : constant_coeff ℚ (exp ℚ) = 1 := rfl
+
+theorem bernoulli_power_series :
+(power_series.mk (λ n, ((bernoulli n) / (nat.factorial n) : ℚ))) * (power_series.exp ℚ - 1) =
+  (X : power_series ℚ) * (exp ℚ) :=
+begin
+  ext n,
+  -- constant coefficient is a special case
+  cases n,
+  { simp only [ring_hom.map_sub, constant_coeff_one, zero_mul, constant_coeff_exp, constant_coeff_X,
+      coeff_zero_eq_constant_coeff, mul_zero, sub_self, ring_hom.map_mul] },
+  rw [coeff_mul, mul_comm X, coeff_succ_mul_X],
+  simp only [coeff_mk, coeff_one, coeff_exp, linear_map.map_sub, factorial,
+    rat.algebra_map_rat_rat], rw nat.sum_antidiagonal_succ',
+  simp, --squeeze_simp hangs
```

Comment:
This is a non-terminal `simp`. Indeed `squeeze_simp` hangs. With a bit of faffing around I managed to remove it (I had to "be the simplifier", which is not something you need to do usually):
```suggestion
  simp only [coeff_mk, coeff_one, coeff_exp, linear_map.map_sub, factorial,
    rat.algebra_map_rat_rat, nat.sum_antidiagonal_succ', if_pos],
  simp only [factorial, prod.snd, one_div, cast_succ, cast_one, cast_mul, ring_hom.id_apply,
    sub_zero, add_eq_zero_iff, if_false, zero_add, one_ne_zero,
    factorial, div_one, mul_zero, and_false, sub_self],
```